internal/nms/helper.go

```diff
@@ -0,0 +1,74 @@
+package nms
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+)
+
+//go:generate mockgen -source=helper.go -package=nms -destination=mock_helper.go
+
+type Helper interface {
+	GetNMS(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error)
+	CreateNMS(ctx context.Context, name string) (*kmmv1beta1.NodeModulesConfig, error)
+	GetNMSModuleEntry(nms *kmmv1beta1.NodeModulesConfig, modNamespace, modName string) *kmmv1beta1.NodeModuleSpec
+	CreateNMSModuleEntry(nms *kmmv1beta1.NodeModulesConfig, modNamespace, modName string) *kmmv1beta1.NodeModuleSpec
```

Comment:
I would remove `NMS` from all methods here; they all belong to an NMC helper.

Reply:
fixed